osquery/tables/system/darwin/kernel_panics.cpp

```diff
@@ -0,0 +1,133 @@
+    {"System model name", "system_model"},
+    {"System uptime in nanoseconds", "uptime"},
+};
+
+void readKernelPanic(const std::string& appLog, Row& r) {
+  r["path"] = appLog;
+  std::string content;
+
+  if (!readFile(appLog, content).ok()) {
+    return;
+  }
+
+  boost::regex rxSpaces("\\s+");
+  auto lines = osquery::split(content, "\n");
+  for (auto it = lines.begin(); it != lines.end(); it++) {
+    auto line = *it;
+    boost::trim(line);
+
+    auto toks = osquery::split(line, ":");
+    if (toks.size() == 0) {
+      continue;
+    }
+
+    auto timeTokens = osquery::split(toks[0], " ");
+    if (timeTokens.size() >= 1 && kDays.count(timeTokens[0]) > 0) {
+      r["time"] = line;
+    }
+
+    if (kRegisters.count(toks[0]) > 0) {
+      auto registerTokens = osquery::split(line, ",");
+      if (registerTokens.size() == 0) {
+        continue;
+      }
+
+      for (auto& tok_ : registerTokens) {
+        auto regHolder = osquery::split(tok_, ":");
+        if (regHolder.size() != 2) {
+          continue;
+        }
+        auto reg = std::move(regHolder[0]);
+        auto val = std::move(regHolder[1]);
+        if (reg.size() > 0 && val.size() > 0) {
+          std::string regLine = reg + ":" + val;
+          r["registers"] += (r["registers"].empty()) ? std::move(regLine)
+                                                     : " " + std::move(regLine);
+        }
+      }
+    } else if (boost::starts_with(toks[0], "last loaded kext at") &&
+               toks.size() == 2) {
+      r["last_loaded_module"] = boost::regex_replace(toks[1], rxSpaces, " ");
+    } else if (boost::starts_with(toks[0], "last unloaded kext at") &&
+               toks.size() == 2) {
+      r["last_unloaded_module"] = boost::regex_replace(toks[1], rxSpaces, " ");
+    } else if (boost::starts_with(toks[0], "Backtrace") &&
+               std::next(it) != lines.end()) {
+      r["frame_backtrace"] = *(std::next(it));
+    } else if (boost::starts_with(toks[0], "Kernel Extensions in backtrace") &&
+               std::next(it) != lines.end()) {
+      r["module_backtrace"] = *(std::next(it));
+    } else if (boost::starts_with(toks[0], "Mac OS version") &&
+               std::next(it) != lines.end()) {
+      r["os_version"] = *(std::next(it));
+    } else if (boost::starts_with(toks[0], "Kernel version") &&
+               std::next(it) != lines.end()) {
+      r["kernel_version"] = *(std::next(it));
+    } else if (kKernelPanicKeys.count(toks[0]) != 0 && toks.size() == 2) {
+      r[kKernelPanicKeys.at(toks[0])] = toks[1];
+    }
+  }
+}
+
+QueryData genKernelPanics(QueryContext& context) {
+  QueryData results;
+
+  if (context.constraints["uid"].notExistsOrMatches("0")) {
+    std::vector<std::string> files;
+    if (listFilesInDirectory(kDiagnosticReportsPath, files)) {
+      for (const auto& lf : files) {
+        if (alg::ends_with(lf, ".panic")) {
+          Row r;
+          readKernelPanic(lf, r);
```

Comment:
This looks odd, why not pass the `QueryResults` as a reference to `readKernelPanic`. Then construct your `Row` within that's methods scope and push back before returning.